components/massmint/mintingHelpers.ts

```diff
@@ -0,0 +1,164 @@
+import { CreatedNFT, Interaction } from '@kodadot1/minimark/v1'
+import { TokenToList, TokenToMint } from '@/composables/transaction/types'
+import useSubscriptionGraphql from '@/composables/useSubscriptionGraphql'
+import { Ref } from 'vue'
+import { toNFTId } from '../rmrk/service/scheme'
+export const subscribeToCollectionUpdates = (collectionId: string) => {
+  const collectionUpdated = ref(false)
+  const numOfNftsInCollections = ref<number>()
+
+  useSubscriptionGraphql({
+    query: `  collection: collectionEntityById(id: "${collectionId}") {
+        id
+        name
+        nfts(
+          orderBy: [updatedAt_DESC]
+          where: { burned_eq: false }
+        ) {
+          id
+          name
+          sn
+    
+        }
+      }`,
+    onChange: ({ data }) => {
+      const collection = data.collection
+      const currentNumberOfNts = collection.nfts.length
+      if (numOfNftsInCollections.value === undefined) {
+        numOfNftsInCollections.value = currentNumberOfNts
+        return
+      }
+
+      if (currentNumberOfNts > numOfNftsInCollections.value) {
+        collectionUpdated.value = true
+      }
+    },
+  })
+
+  return collectionUpdated
+}
+
+export const listForSell = (mintedNFts: TokenToList[]) => {
+  const { blockNumber, transaction, isLoading, status } = useTransaction()
+  const { urlPrefix } = usePrefix()
+  const { $i18n } = useNuxtApp()
+
+  transaction({
+    interaction: Interaction.LIST,
+    urlPrefix: urlPrefix.value,
+    token: mintedNFts,
+    successMessage: $i18n.t('transaction.price.success') as string,
+    errorMessage: $i18n.t('transaction.price.error') as string,
+  })
+  return {
+    blockNumber,
+    isLoading,
+    status,
+  }
+}
+
+export const getListForSellItems = (
+  createdNFTs: CreatedNFT[],
+  tokens: TokenToMint[],
+  blockNumber: string
+) => {
+  const { $consola } = useNuxtApp()
+  return createdNFTs
+    .map((nft: CreatedNFT) => {
+      const matchingToken = tokens.find(
+        (token) =>
+          token.name === nft?.name &&
+          token.selectedCollection?.id === nft.collection
+      )
+
+      if (matchingToken === undefined) {
+        $consola.error('matching token not found', nft)
+        return undefined
+      }
+      if (matchingToken.price === undefined) {
+        return undefined
+      }
+      const nftId = toNFTId(nft, blockNumber)
```

Comment:
@vikiival 
```toNFTId``` function expects ```instance``` property on the nft
but ```createdNft``` imported from ```@kodadot1/minimark/v2```doesn't have that.

does that mean it will not work on RMRK2